app/controllers/idv/verify_info_controller.rb

```diff
@@ -74,5 +120,284 @@ def current_flow_step_counts
+      )
+
+      if form_response.success?
+        response = check_ssn
+        form_response = form_response.merge(response)
+      end
+      summarize_result_and_throttle_failures(form_response)
+      delete_async
+
+      if form_response.success?
+        idv_session.resolution_successful = true
+        redirect_to idv_phone_url
+      else
+        idv_session.resolution_successful = false
+      end
+
+      analytics.idv_doc_auth_verify_proofing_results(**form_response.to_h)
+    end
+
+    def summarize_result_and_throttle_failures(summary_result)
+      if summary_result.success?
+        add_proofing_components
+        summary_result
+      else
+        idv_failure(summary_result)
+      end
+    end
+
+    def add_proofing_components
+      ProofingComponent.create_or_find_by(user: current_user).update(
+        resolution_check: Idp::Constants::Vendors::LEXIS_NEXIS,
+        source_check: Idp::Constants::Vendors::AAMVA,
+      )
+    end
+
+    def load_async_state
+      dcs_uuid = idv_session.verify_info_step_document_capture_session_uuid
+      dcs = DocumentCaptureSession.find_by(uuid: dcs_uuid)
+      return ProofingSessionAsyncResult.none if dcs_uuid.nil?
+      return ProofingSessionAsyncResult.missing if dcs.nil?
+
+      proofing_job_result = dcs.load_proofing_result
+      return ProofingSessionAsyncResult.missing if proofing_job_result.nil?
+
+      proofing_job_result
+    end
+
+    def delete_async
+      idv_session.verify_info_step_document_capture_session_uuid = nil
+    end
+
+    def idv_result_to_form_response(
+      result:,
+      state: nil,
+      state_id_jurisdiction: nil,
+      state_id_number: nil,
+      extra: {}
+    )
+      state_id = result.dig(:context, :stages, :state_id)
+      if state_id
+        state_id[:state] = state if state
+        state_id[:state_id_jurisdiction] = state_id_jurisdiction if state_id_jurisdiction
+        state_id[:state_id_number] = redact(state_id_number) if state_id_number
+      end
+
+      FormResponse.new(
+        success: result[:success],
+        errors: result[:errors],
+        extra: extra.merge(proofing_results: result[:context]),
+      )
+    end
+
+    def log_idv_verification_submitted_event(success: false, failure_reason: nil)
+      pii_from_doc = pii || {}
+      irs_attempts_api_tracker.idv_verification_submitted(
+        success: success,
+        document_state: pii_from_doc[:state],
+        document_number: pii_from_doc[:state_id_number],
+        document_issued: pii_from_doc[:state_id_issued],
+        document_expiration: pii_from_doc[:state_id_expiration],
+        first_name: pii_from_doc[:first_name],
+        last_name: pii_from_doc[:last_name],
+        date_of_birth: pii_from_doc[:dob],
+        address: pii_from_doc[:address1],
+        ssn: pii_from_doc[:ssn],
+        failure_reason: failure_reason,
+      )
+    end
+
+    def redact(text)
+      text.gsub(/[a-z]/i, 'X').gsub(/\d/i, '#')
+    end
```

Comment:
I think this is the 3rd time we have copy-pasted this method definition ([one](https://github.com/18F/identity-idp/blob/cd865f322a457388c6306d2cd53315689eb78e31/app/services/idv/steps/verify_base_step.rb#L308), [two](https://github.com/18F/identity-idp/blob/cd865f322a457388c6306d2cd53315689eb78e31/app/forms/new_phone_form.rb#L142)), I think that makes it a good candidate to be unified by a helper. like `StringRedacter.redact_alphanumeric` or something

Reply:
How is that usually done? Separate ticket + small PR? I looked at the existing helpers and looks like we would need a new one rather than adding to an existing one. Maybe something a little more general than StringRedacter, like PrivacyHelper?

Reply:
It depends on the complexity. To me, it looks like all 3 are the exact same method? And it would be a straightforward static method extraction, so I would consider doing it in this PR (because the additional overhead is almost the same as copy-pasting like we've done here). But if needed, we could file a ticket and follow up, which makes sense in more complicated refactors

As far as naming, I don't feel that strongly, `PrivacyHelper` seems fine, in general I feel like `xyzHelper` classes are not always super clear, I'd name it after what it does (redact strings), but to me the more important bit is extracting the shared code

Reply:
Fixed by adding a StringRedacter concern. Will shift over the other places it's used separately from this PR. Also noting that VerifyInfoController will eventually replace VerifyBaseStep, so one of the uses will go away.

Tried adding a Helper first, but helpers are expected to be used in Views. Then tried adding it in lib, then settled on a Concern since that's what's used in Controllers.